### ISSUE TITLE
Station sometimes fails to create the the azuread_application

### ISSUE THREAD

**Issue by ghost** [blinqas/station#103]
Sometimes when creating an azureAD application with station you will get an error similar to this: 
```
│ Error: Could not create application
│ 
│   with module.station-applications.module.applications["minimum"].azuread_application.app,
│   on ../application/azuread_application.tf line 1, in resource "azuread_application" "app":
│    1: resource "azuread_application" "app" {
│ 
│ ApplicationsClient.BaseClient.Post(): unexpected status 404 with OData
│ error: Request_ResourceNotFound: Resource
│ '98d8bd02-91ca-4d8b-b861-d1df80c48c99' does not exist or one of its queried
│ reference-property objects are not present.

│ Error: Could not create application
│ 
│   with module.station-applications.module.applications["maximum"].azuread_application.app,
│   on ../application/azuread_application.tf line 1, in resource "azuread_application" "app":
│    1: resource "azuread_application" "app" {
│ 
│ ApplicationsClient.BaseClient.Post(): unexpected status 404 with OData
│ error: Request_ResourceNotFound: Resource
│ '98d8bd02-91ca-4d8b-b861-d1df80c48c99' does not exist or one of its queried
│ reference-property objects are not present.

``` 
This is something that just happens from time to time and we should figure out why this is happening. You can try to provoke the issue by running the test for the application (`terraform apply -target module.station-applications`). The resource `'98d8bd02-91ca-4d8b-b861-d1df80c48c99` is a reference to the principal_object_id.

**Comment by SanderBlom** [comment]
**Possible Cause for Azure AD App Creation Glitch**

I think I've pinpointed the intermittent errors with the AD application creation in Station. It seems to be a race condition in code. The issue arises when certain resources aren't created by the time we attempt to create the Azure AD app, leading to those 404 errors. Interestingly, I've found that running `terraform apply` a second time often resolves the issue, which further points to this being a timing-related problem.
